fix(ui): Add ellipsis for overflowing text in stream filter.

VANSH3104 opened the pull request:
<!-- Describe your pull request here.-->
Fixes ellipsis for overflowing text in stream filter. discussed in #33183
Fixes: <!-- Issue link, or clear description.-->
before 
![0](https://github.com/user-attachments/assets/f1c40bb7-3218-448a-b2b7-bbae5b60c4e4)
after
![1](https://github.com/user-attachments/assets/0bd96eef-af68-4a82-99ce-d1633be2d105)

<!-- If the PR makes UI changes, always include one or more still screenshots to demonstrate your changes. If it seems helpful, add a screen capture of the new functionality as well.

Tooling tips: https://zulip.readthedocs.io/en/latest/tutorials/screenshot-and-gif-software.html
-->

**Screenshots and screen captures:**

<details>
<summary>Self-review checklist</summary>

<!-- Prior to submitting a PR, follow our step-by-step guide to review your own code:
https://zulip.readthedocs.io/en/latest/contributing/code-reviewing.html#how-to-review-code -->

<!-- Once you create the PR, check off all the steps below that you have completed.
If any of these steps are not relevant or you have not completed, leave them unchecked.-->

- [x] [Self-reviewed](https://zulip.readthedocs.io/en/latest/contributing/code-reviewing.html#how-to-review-code) the changes for clarity and maintainability
      (variable names, code reuse, readability, etc.).

Communicate decisions, questions, and potential concerns.

- [ ] Explains differences from previous plans (e.g., issue description).
- [ ] Highlights technical choices and bugs encountered.
- [ ] Calls out remaining decisions and concerns.
- [ ] Automated tests verify logic where appropriate.

Individual commits are ready for review (see [commit discipline](https://zulip.readthedocs.io/en/latest/contributing/commit-discipline.html)).

- [ ] Each commit is a coherent idea.
- [ ] Commit message(s) explain reasoning and motivation for changes.

Completed manual review and testing of the following:

- [x] Visual appearance of the changes.
- [x] Responsiveness and internationalization.
- [x] Strings and tooltips.
- [x] End-to-end functionality of buttons, interactions and flows.
- [ ] Corner cases, error conditions, and easily imagined bugs.
</details>

VANSH3104 commented:
@amanagr I placed another pr as you told me it handles only stream filter overflowing

whilstsomebody commented:
The code works for me.

whilstsomebody commented:
I recommend not using "fix" as a preifx, use "stream_filter" either.
It would be good to provide a description for the commit.
Complete the self review checklist.

amanagr commented:
Code changes look good to me. I tested too.

timabbott commented on web/styles/zulip.css:
```diff
@@ -1965,6 +1965,11 @@ body:not(.hide-left-sidebar) {
+    .stream-name {
+        overflow: hidden;
+        text-overflow: ellipsis;
+    }
```
This class name is a bad choice; it overlaps with other styles in the app, like this one in `left-sidebar.css`

``` 
.conversation-partners-list,                                                                            
.stream-name {                                                                                          
    white-space: nowrap;                                                                                
    overflow-x: hidden;                                                                                 
    overflow-x: clip;                                                                                   
    text-overflow: ellipsis;                                                                            
    padding: 0 var(--left-sidebar-before-unread-count-padding) 0 0;                                     
}                                                                                                       
```

Adding this class to `inline_decorated_stream_name` will end up affecting several other UI components, too.

timabbott commented on web/templates/inline_decorated_stream_name.hbs:
```diff
@@ -4,5 +4,5 @@
 {{~ else ~}}
-<i class="zulip-icon zulip-icon-hashtag stream-privacy-type-icon" {{#if show_colored_icon}}style="color: {{stream.color}}"{{/if}} aria-hidden="true"></i> {{stream.name ~}}
+<i class="zulip-icon zulip-icon-hashtag stream-privacy-type-icon" {{#if show_colored_icon}}style="color: {{stream.color}}"{{/if}} aria-hidden="true"></i> <span class="stream-name">{{stream.name ~}}</span>
 {{~/if~}}
```
This code is very wrong:
- You only added the class to one of 3 cases, so private or web-public channels will have the bug.
- It applies to many other components that share this template, and I don't see any effort to evaluate whether that will improve or break them.

VANSH3104 replied:
@timabbott ok i will correct it

timabbott commented:
Dropping integration review, as this needs significant reworking.

VANSH3104 commented:
@timabbott i change it for all private and public streams and also change the name 
![Screenshot from 2025-02-12 13-33-50](https://github.com/user-attachments/assets/c12a355e-c3a0-4c0c-bed3-353b6f67b80a)

VANSH3104 commented:
@timabbott @amanagr  i saw the same issue in lots off places too in hover and sidebars 
![Screenshot from 2025-02-12 13-57-32](https://github.com/user-attachments/assets/3caa15fa-238e-4894-9143-ccc8a9f389c9)
![Screenshot from 2025-02-12 13-57-20](https://github.com/user-attachments/assets/bf2cd515-fd11-493e-8c19-e313ae4fd2e8)
![Screenshot from 2025-02-12 13-57-06](https://github.com/user-attachments/assets/9e406d88-e61d-4adc-b7f5-10a5a792f81b)
![Screenshot from 2025-02-12 13-56-50](https://github.com/user-attachments/assets/afd7c0b2-96f0-4e65-bc56-b3e1b4eac8f9)
can I work on them too

alya commented:
@amanagr could you please add this one to your post-release review queue?

@VANSH3104 You can work on improving your commit message in the meantime, and describing carefully how you addressed @timabbott 's feedback above.

VANSH3104 commented:
@alya Thanks for the feedback! I’ve updated the commit message.
- @timabbott mentioned that my code conflicted with existing CSS classes in other files, so I updated the class name to avoid conflicts.
- Initially, the fix only applied to public streams, so I updated it to work for all types of streams (private, public, and web-public).
- Made sure the changes don’t break other UI components.
Let me know if anything else needs improvement!

timabbott commented:
@karlstolley is this still useful following d5822a04be353e709c48a602db3d14b68acb4211? I'm wondering if we should close this and/or the issue it's associated with.
